src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Private/JavaScriptMarshalImpl.Invoke.cs

```diff
@@ -27,7 +27,7 @@ internal static unsafe void InvokeJSFunctionByName<T1>(string functionName, T1 a
             var data = stackalloc JavaScriptMarshallerArg[argsCount + 2];
```

Comment:
it might be nice to find a way to abstract this so the array is [argsCount] and the special slots have names, i.e. something like:
```csharp
var record = new JavascriptMarshalRecord {
  pData = stackalloc JavaScriptMarshalArg[argsCount]
};
DynamicMarshalToJs(ref record, ref signature);
```
and then record would have fields for the slots that are always there, like the result and exception.

Reply:
done